### PR TITLE
fix: disable confirm button

### PR DESCRIPTION
**Pull request by henriettemoe**
- Makes the confirm button of the "ChooseCollectionDialog" disabled if there are no changed to submit. This also concerns the dialog when creating a new collection, where the confirm button will be disabled until the text field is filled in.
- Adds disabled and hover styling to the "default" button, and hover styling of "secondary" button
- Adjusts some styling related to the dialog
 
Related to Trello-issue: https://trello.com/c/6MU1MTfo/450-du-f%C3%A5r-opp-beskjed-om-at-endringer-lagres-uten-at-det-er-lagret

Screenshot:
Skjermbilde 2024-09-11 kl  17 12 10

### CODE REVIEW

File: common/components/ConfirmationDialog/ConfirmationDialog.tsx [hunk 1]
```diff
@@ -30,7 +32,7 @@ const ConfirmationDialog = ({
         <Button variant="secondary" onClick={onCancel}>
           {cancel}
         </Button>
-        <Button variant="default" onClick={onConfirm}>
+        <Button variant="default" disabled={disableConfirm} onClick={onConfirm}>
```
Comment (MichalRLoj) [inline common/components/ConfirmationDialog/ConfirmationDialog.tsx:35]: Suggestion add a tooltip for the user why the button is disabled for example "Please make changes " to indicate what the user is doing wrong or indication that the text has been change like "Changes have been made"

File: common/components/ConfirmationDialog/ConfirmationDialog.tsx [hunk 2]
```diff
@@ -30,7 +32,7 @@ const ConfirmationDialog = ({
         <Button variant="secondary" onClick={onCancel}>
           {cancel}
         </Button>
-        <Button variant="default" onClick={onConfirm}>
```
Comment (MichalRLoj) [inline common/components/ConfirmationDialog/ConfirmationDialog.tsx:33]: Also onclick still works even though the button is grayed out. Not sure if that is intended

File: common/hooks/useCollectionDialog.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import { useState, useCallback, useEffect } from "react";
 import { useDialogContext } from "common/hooks/useDialogContext";
 import { useMyCollections } from "common/hooks/useMyCollections";
-import { Option } from "common/types/types";
+import { Collection, Option } from "common/types/types";
```
Comment (MichalRLoj) [inline common/hooks/useCollectionDialog.ts:6]: Suggestion, add a cached  state for the initial selected options, if the user closes the dialog something like  ` const [initialSelectedOptions, setInitialSelectedOptions] = useState<string[]>([]);`  and in the handleCancel add the   `setSelectedOptions(initialSelectedOptions); `  this is to its original state if the dialog is canceled

**Review by MichalRLoj — CHANGES_REQUESTED** [review]

File: common/components/ConfirmationDialog/ConfirmationDialog.tsx (hunk 2 above)
Comment (henriettemoe) [reply common/components/ConfirmationDialog/ConfirmationDialog.tsx:33]: It doesn't work when it's disabled. Have you tested locally and encountered that it still works even though it's disabled?

File: common/hooks/useCollectionDialog.ts (hunk 3 above)
Comment (henriettemoe) [reply common/hooks/useCollectionDialog.ts:6]: Currently, when canceling the dialog it does get back to it's original state. If you see room for improvement on how this is implemented, we can perhaps do it in a different PR?

File: common/components/ConfirmationDialog/ConfirmationDialog.tsx (hunk 1 above)
Comment (henriettemoe) [reply common/components/ConfirmationDialog/ConfirmationDialog.tsx]: Tooltip is now added. There are two different versions for "Choose a collection". If there is no preselected collection (the word is not added to a collection prior opening the dialog) then the tooltip says "Choose a collection". But if there is any preselected collection the tooltip says "The word is already added to the selected collections".

Skjermbilde 2024-09-16 kl  17 39 00
Skjermbilde 2024-09-16 kl  17 39 12

**Review by MichalRLoj — APPROVED** [review]

**Review by MichalRLoj — APPROVED** [review]

**Pull request merged**